keras/src/backend/common/remat_scope.py

```diff
@@ -0,0 +1,127 @@
+from keras.src.backend.common import global_state
+
+
+class RematScope:
+    """A context manager for enabling rematerialization in Keras.
+
+    Rematerialization (gradient checkpointing) trades memory for computation by
+    recomputing intermediate activations during the backward pass. This is
+    particularly useful for training large models or large batch sizes within
+    limited memory constraints.
+
+    Args:
+        mode: Rematerialization mode to apply.
+            Options:
+            - "full": Apply rematerialization globally to all supported
+              operations.
+            - "activations": Apply rematerialization only to activation layers.
+            - "larger_than": Apply rematerialization to layers with output sizes
+              larger than `output_size_threshold`.
+            - "list_of_layers": Apply rematerialization to a specific list of
```

Comment:
Based on the arguments it really sounds like this should have been a layer constructor arg... what was the motivation for making it a scope?

Reply:
Users can now specify a list of layer names in the remat scope to apply rematting selectively to those layers. This provides fine-grained control over which layers are rematted in a model. This was a mode added based on feedback from customers.